runtime/src/chpl-cache.c

```diff
@@ -2788,10 +2814,13 @@ void chpl_cache_comm_put(void* addr, c_nodeid_t node, void* raddr,
   //printf("put len %d node %d raddr %p\n", (int) len * elemSize, node, raddr);
   struct rdcache_s* cache = tls_cache_remote_data();
   if (size_merits_direct_comm(cache, size)) {
+    cache_lock(cache);
```

Comment:
Any reason to not throw the lock/unlock into cache_invalidate?

Reply:
Yes, there are two reasons.
1. cache_invalidate is not a public/API method and generally this PR takes the approach of adding the locking to the public methods only
2. `chpl_cache_comm_getput_unordered` actually calls cache_invalidate twice within the same "lock"ed region.

Reply:
> 1. cache_invalidate is not a public/API method and generally this PR takes the approach of adding the locking to the public methods only

I don't think I understand the benefit of adding to only the public APIs. Mind expanding on that a bit?

> chpl_cache_comm_getput_unordered actually calls cache_invalidate twice within the same "lock"ed region.

Saw that, but I figured the overhead of this "lock" is negligible so that didn't bother me. If it was an atomic lock or something I'd want to avoid duplicate calls.

I'm fine leaving it as-is if you prefer it this way, just seemed like we could eliminate some of the calls.

Reply:
> I don't think I understand the benefit of adding to only the public APIs. Mind expanding on that a bit?

Oh, it's just that the places the lock / unlock calls occur are consistent. Not a big deal but this kind of thing aids understanbility.

> Saw that, but I figured the overhead of this "lock" is negligible so that didn't bother me. 

Yep, I agree about that too.

Reply:
Sounds good